changelog/fragments/upgrade-kb.yaml

```diff
@@ -0,0 +1,57 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      Upgrade controller-runtime (sigs.k8s.io/controller-runtime) dependency version
+      from `v0.6.2` to `v0.6.3`. More info: https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.6.3
+
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "change"
+
+    # Is this a breaking change?
+    breaking: false
+
+    # Migration can be defined to automatically add a section to
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: Upgrade controller-runtime version
+      body: >
+        In the `go.mod` file replace `sigs.k8s.io/controller-runtime v0.6.2`
+        with `sigs.k8s.io/controller-runtime v0.6.3` and then run `go mod tidy`.
+  - description: >
+      Upgrade the Kubebuilder CLI dependecy version from `f7a3b65dd250` to `c993a2a221fe`.
+      More info: https://github.com/kubernetes-sigs/kubebuilder/compare/f7a3b65dd250..c993a2a221fe
+
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "change"
+
+    # Is this a breaking change?
+    breaking: false
+
+    # Migration can be defined to automatically add a section to
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: Fix the `CRDDirectoryPath` path in the `suite_test.go` files
+      body: >
+        For Golang based-operator projects, fix `CRDDirectoryPath` value in the `controllers/<group>/suite_test.go` files
+        when the project has multi-group support. More info: https://github.com/kubernetes-sigs/kubebuilder/issues/1665
+
+        If your project has multi-group support, then replace
+        `CRDDirectoryPaths: []string{filepath.Join("..", "config", "crd", "bases")},` with
+        `CRDDirectoryPaths: []string{filepath.Join("..", "..", "config", "crd", "bases")},`
+        in the `suite_test.go` files found in the `controllers/<group>/` directories.
+
+        **NOTE** To check if your project has multigroup support ensure that
+        your PROJECT file has the `multigroup: true` attribute.
```

Comment:
Hmm... this one actually seems required to fix a bug in the scaffolding, so I agree that its good to document what should change in existing projects.

Two things that I think all migrations need:
1. Context about what set of operators/projects could make use of this migration. (e.g. this only applies to multi-group Go operators)
2. What happens if I don't perform this migration (e.g. I'm guessing e2e tests fail to correctly deploy the CRDs, and therefore probably fail entirely)

Reply:
@joelanford, 

I am applying the @estroz suggestion. I think we do not need to be too much verbose. For further info, users can check the issue in the kb.

Reply:
done.